File: pytools/examples/lua53/server/protocols/AckRoleRandNameOk.lua
AckRoleRandNameOk = {}
setmetatable(AckRoleRandNameOk, {__index = _G})
local _ENV = AckRoleRandNameOk


function new()
	local tab_obj = {}
	setmetatable(tab_obj, {__index = AckRoleRandNameOk})
	tab_obj:init()
	return tab_obj
end

function init(self)
	self.uname = 0
end


function encode(self)
	pack = packet.new()
	pack:writeString(self.uname)
	return pack:encode(Msg.P_ACK_ROLE_RAND_NAME_OK)
end


function setUname(self, uname)
	self.uname = uname
end

File: pytools/data/data.code.golang.server/ack_role_rand_name_ok.go
package proto

import (
	"packet"
)

type AckRoleRandNameOk struct {
	uname                    string
}

func (this *AckRoleRandNameOk) Encode() []byte {
	pack := packet.NewWriteBuff(64)

	pack.WriteString(this.uname)

	return pack.Encode(P_ACK_ROLE_RAND_NAME_OK)
}

func (this *AckRoleRandNameOk) SetUname(uname string) {
	this.uname = uname
}
